pkg/apiserver/rest/dualwriter.go

```diff
@@ -145,51 +145,44 @@ func SetDualWritingMode(
 		}
 	}
 
-	// Handle transitions to the desired mode.
-	switch {
-	case cfg.Mode == Mode2 || cfg.Mode == Mode1:
-		// Directly set the mode for Mode1 and Mode2.
-		currentMode = cfg.Mode
-		if err := kvs.Set(ctx, cfg.Kind, fmt.Sprint(currentMode)); err != nil {
+	isUpgradeToReadUnifiedMode := currentMode < Mode3 && cfg.Mode >= Mode3
+	if !isUpgradeToReadUnifiedMode {
```

Comment:
given that the if-blocks looks the same, could we do something like this? (and then remove the if-block from [here](https://github.com/grafana/grafana/pull/106692/files#diff-e4fe089a41bba5a9ac7348794b35cb4c8edc29a0650f2e51bf8760f4d6492d36R157)
```suggestion
	if !isUpgradeToReadUnifiedMode || cfg.SkipDataSync {
```

Reply:
We could for sure. I just wanted the case we are covering to be more obvious. Let me address that.

Reply:
I thought about it again 🙂  I think it would a bit more “beneficial” to let them stay as is? It is not always easy to understand the cases clearly around the codebase, so I would argue whether it is better to keep it as more explicit and keep the case separate? WDYT?